src/main/java/net/mcreator/element/types/Achievement.java

```diff
@@ -96,4 +102,30 @@ public boolean hasRewards() {
 		};
 	}
 
+	@Override public Collection<String> getUsedElementNames() {
+		List<String> elements = new ArrayList<>();
+		if (rewardFunction != null && !rewardFunction.equals("No function"))
+			elements.add("CUSTOM:" + rewardFunction);
+		rewardLoot.forEach(e -> elements.add("CUSTOM:" + e));
+		rewardRecipes.forEach(e -> elements.add("CUSTOM:" + e));
+		return elements;
+	}
+
+	@Override public Collection<? extends MappableElement> getUsedElementMappings() {
+		if (disableDisplay)
+			return Collections.emptyList();
+		List<MappableElement> elements = new ArrayList<>();
+		elements.add(achievementIcon);
+		if (!parent.getUnmappedValue().equals("No parent: root"))
+			elements.add(parent);
+		return elements;
+	}
+
+	@Override public Collection<String> getTextures(TextureType type) {
+		return type == TextureType.SCREEN ? Collections.singletonList(background) : Collections.emptyList();

```

Comment:
Background can also be "Default", which is no dependency and thus no reference should be passed. I think you should re-check MEs again

Reply:
Similar in what sense?

Reply:
Similar in the sense that MEs are reporting usage of elements or resources that are not true references. "Default" background is not a valid reference

Reply:
✔️ Fixed